sql-plugin/src/main/scala/com/nvidia/spark/rapids/PrioritySemaphore.scala

```diff
@@ -56,17 +54,28 @@ class PrioritySemaphore[T](val maxPermits: Int)(implicit ordering: Ordering[T])
 
   def acquire(numPermits: Int, priority: T): Unit = {
     lock.lock()
+    val condition = lock.newCondition()
+    val info = ThreadInfo(priority, condition, numPermits)
```

Comment:
This should be moved into the `else` below so we don't create objects we don't need.  Requires moving the `try` block into the `else` as well.